tools/stress-cluster/cluster/kubernetes/stress-test-addons/templates/stress-test-static-secret-provider.tpl

```diff
@@ -0,0 +1,37 @@
+{{- define "stress-test-addons.static-secrets" -}}
+{{- /* The subscriptionConfig key allows a user to use a custom set of subscription+creds uploaded manually to the static keyvault */}}
+{{- /* This template finds all unique subscription configs in scenarios and creates a single secret provider for each */}}
+{{- $global := . }}
+{{- $subChart := get $global.Values "stress-test-addons" }}
+{{- $subConfigs := list (get $subChart.subscription $subChart.env) }}
+{{- range $global.Values.scenarios }}
+{{- $subConfigs = append $subConfigs (coalesce .subscriptionConfig "") }}
+{{- end }}
+{{- $subConfigs = compact $subConfigs | uniq }}
+{{- range $subConfigs }}
+---
+apiVersion: secrets-store.csi.x-k8s.io/v1
```

Comment:
This file/resource is pretty similar to the yaml file deleted in this PR (I can't get git to show a diff as a moved file). The main changes is the SecretProviderClass is specified in a loop for all possible `subscriptionConfiguration` values found in the matrix, so that we download all possible secrets referenced.

Additionally, the SecretProviderClass resource now has a dynamic naming convention that includes the secret name. This way individual scenario tests just reference their respective subscription config from the new constant `{{ .Stress.SubscriptionConfig }}` via the template, so they only load the appropriate config and don't clobber with any of the others.